File: src/ios/Theme.swift
import UIKit

@objc(Theme)
class Theme: CDVPlugin {

  // MARK: - setStyle

  @objc(setStyle:)
  func setStyle(_ command: CDVInvokedUrlCommand) {
    guard let style = command.arguments.first as? String else {
      let result = CDVPluginResult(status: CDVCommandStatus_ERROR, messageAs: "Missing style argument")
      commandDelegate.send(result, callbackId: command.callbackId)
      return
    }

    guard #available(iOS 13, *) else {
      let result = CDVPluginResult(status: CDVCommandStatus_OK)
      commandDelegate.send(result, callbackId: command.callbackId)
      return
    }

    let interfaceStyle: UIUserInterfaceStyle
    switch style.lowercased() {
    case "light":
      interfaceStyle = .light
    case "dark":
      interfaceStyle = .dark
    default:
      interfaceStyle = .unspecified
    }

    DispatchQueue.main.async { [weak self] in
      self?.viewController.view.window?.overrideUserInterfaceStyle = interfaceStyle
      let result = CDVPluginResult(status: CDVCommandStatus_OK)
      self?.commandDelegate.send(result, callbackId: command.callbackId)
    }
  }

  // MARK: - getStyle

  @objc(getStyle:)
  func getStyle(_ command: CDVInvokedUrlCommand) {
    guard #available(iOS 13, *) else {
      let result = CDVPluginResult(status: CDVCommandStatus_OK, messageAs: "system")
      commandDelegate.send(result, callbackId: command.callbackId)
      return
    }

    DispatchQueue.main.async { [weak self] in
      let style: String
      switch self?.viewController.view.window?.overrideUserInterfaceStyle {
      case .light:
        style = "light"
      case .dark:
        style = "dark"
      default:
        style = "system"
      }
      let result = CDVPluginResult(status: CDVCommandStatus_OK, messageAs: style)
      self?.commandDelegate.send(result, callbackId: command.callbackId)
    }
  }
}

File: www/Theme.js
var exec = require('cordova/exec');

/**
 * Switch iOS overrideUserInterfaceStyle.
 *
 * @namespace cordova.plugins.Theme
 */
var Theme = {

  /**
   * Set the UI style.
   *
   * @param {string} style - 'light', 'dark', or 'auto'
   * @param {Function} [success]
   * @param {Function} [error]
   */
  setStyle: function(style, success, error) {
    exec(success, error, 'Theme', 'setStyle', [style]);
  },

  /**
   * Get the current UI style override.
   *
   * @param {Function} success - called with 'light', 'dark', or 'auto'
   * @param {Function} [error]
   */
  getStyle: function(success, error) {
    exec(success, error, 'Theme', 'getStyle', []);
  },
};

module.exports = Theme;
